### PR TITLE
DIT-1020 Suppress Case Modifications

### PR DESCRIPTION
**Pull request by ed0906**

### CODE REVIEW

File: app/uk/gov/hmrc/tariffclassificationfrontend/controllers/SuppressCaseController.scala [hunk 1]
```diff
@@ -38,35 +38,45 @@ class SuppressCaseController @Inject()(verify: RequestActions,
...
+          }
+        )
+      case None =>
+        form.bindFromRequest().fold(
+          errors => {
+            val formWithErrors = errors.withError("email", "You must upload an email")
+            getCaseAndRenderView(reference, c => successful(views.html.suppress_case(c, formWithErrors)))
+          },
+          note => {
+            val formWithErrors = form.fill(note).withError("email", "You must upload an email")
```
Comment (SteveSugden) [inline app/uk/gov/hmrc/tariffclassificationfrontend/controllers/SuppressCaseController.scala]: You must add a note?

Comment (ed0906) [reply app/uk/gov/hmrc/tariffclassificationfrontend/controllers/SuppressCaseController.scala]: Already done as part of the form validation, annoyingly it cant handle files so we have to do that separately

Comment (SteveSugden) [reply app/uk/gov/hmrc/tariffclassificationfrontend/controllers/SuppressCaseController.scala]: Gotcha

File: test/unit/uk/gov/hmrc/tariffclassificationfrontend/service/CasesService_SuppressCaseSpec.scala [hunk 2]
```diff
@@ -57,34 +57,62 @@ class CasesService_SuppressCaseSpec extends UnitSpec with MockitoSugar with Befo
...
+      attachmentUpdating.map(_.id) shouldBe Some("id")
+      attachmentUpdating.map(_.public) shouldBe Some(false)
+      attachmentUpdating.flatMap(_.operator) shouldBe Some(operator)
 
       val eventCreated = theEventCreatedFor(connector, caseUpdated)
       eventCreated.operator shouldBe Operator("operator-id", Some("Billy Bobbins"))
       eventCreated.details shouldBe CaseStatusChange(CaseStatus.NEW, CaseStatus.SUPPRESSED)
     }
 
+    "update update on attachment upload failure" in {
```
Comment (aaroncarroll82) [inline test/unit/uk/gov/hmrc/tariffclassificationfrontend/service/CasesService_SuppressCaseSpec.scala]: Is this test unfinished? The name of it makes no sense, and nothing is being asserted.

File: test/unit/uk/gov/hmrc/tariffclassificationfrontend/controllers/SuppressCaseControllerSpec.scala [hunk 3]
```diff
@@ -106,52 +108,59 @@ class SuppressCaseControllerSpec extends WordSpec with Matchers with UnitSpec
...
       val result: Result =
         await(controller(caseWithStatusNEW).postSuppressCase("reference")
-             (newFakePOSTRequestWithCSRF(fakeApplication).withFormUrlEncodedBody("state" -> "false")))
+        (newFakePOSTRequestWithCSRF(fakeApplication).withBody(aMultipartFileWithParams())))
 
-      status(result) shouldBe Status.SEE_OTHER
-      locationOf(result) shouldBe Some("/tariff-classification/cases/reference/suppress/contact-info")
+      status(result) shouldBe Status.OK
+      bodyOf(result) should include("Change the status of this case to: Suppressed")
     }
```
Comment (aaroncarroll82) [inline test/unit/uk/gov/hmrc/tariffclassificationfrontend/controllers/SuppressCaseControllerSpec.scala]: Should we also be testing here for return to form on file exceeded max size, and file was unsupported type?

File: app/uk/gov/hmrc/tariffclassificationfrontend/controllers/SuppressCaseController.scala [hunk 4]
```diff
@@ -38,35 +39,45 @@ class SuppressCaseController @Inject()(verify: RequestActions,
...
-      {
-        case true => validateAndRedirect(casesService.suppressCase(_, request.operator).map(c => routes.SuppressCaseController.confirmSuppressCase(reference)))
-        case _ => validateAndRedirect(c => successful(routes.SuppressCaseController.showContactInformation(reference)))
-      }
-    )
+    request.body.file("email").filter(_.filename.nonEmpty).filter(_.contentType.isDefined) match {
+      case Some(file) =>
+        form.bindFromRequest().fold(
+          errors =>
+            getCaseAndRenderView(reference, c => successful(views.html.suppress_case(c, errors))),
```
Comment (aaroncarroll82) [inline app/uk/gov/hmrc/tariffclassificationfrontend/controllers/SuppressCaseController.scala]: I'm facing the same issue on my change - but when you go down this route (you added a valid file but no note), when the page reloads, the file you choose is no longer selected, but no error shows, so then you can't click on confirm successfully until picking a file again.

File: test/unit/uk/gov/hmrc/tariffclassificationfrontend/service/CasesService_SuppressCaseSpec.scala (hunk 2 above)
Comment (ed0906) [reply test/unit/uk/gov/hmrc/tariffclassificationfrontend/service/CasesService_SuppressCaseSpec.scala]: Not sure what happened there, sorted

File: app/uk/gov/hmrc/tariffclassificationfrontend/controllers/SuppressCaseController.scala (hunk 4 above)
Comment (ed0906) [reply app/uk/gov/hmrc/tariffclassificationfrontend/controllers/SuppressCaseController.scala]: I think its got to be like this for security reasons :( We can pre-populate the note but not the file

Comment (aaroncarroll82) [reply app/uk/gov/hmrc/tariffclassificationfrontend/controllers/SuppressCaseController.scala]: Fair enough, it's a minor annoyance at most I guess anyway.

File: test/unit/uk/gov/hmrc/tariffclassificationfrontend/controllers/SuppressCaseControllerSpec.scala (hunk 3 above)
Comment (ed0906) [reply test/unit/uk/gov/hmrc/tariffclassificationfrontend/controllers/SuppressCaseControllerSpec.scala]: I pulled the stuff you did across

**Pull request merged**